File: src/test/java/name/gyger/jmoney/session/SessionServiceTests.java
package name.gyger.jmoney.session;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import javax.transaction.Transactional;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest
@Transactional
public class SessionServiceTests {

    @Autowired
    SessionService sessionService;

    @Test
    public void testBasics() {
        sessionService.initSession();
        assertThat(sessionService.isSessionAvailable()).isTrue();

        sessionService.handleContextRefresh(null);
        assertThat(sessionService.isSessionAvailable()).isTrue();

        Session s = sessionService.getSession();
        assertThat(s).isNotNull();
        assertThat(s.getRootCategory()).isNotNull();
        assertThat(s.getSplitCategory()).isNotNull();
        assertThat(s.getTransferCategory()).isNotNull();
        assertThat(s.getId()).isNotNull();

        sessionService.removeOldSession();
        assertThat(sessionService.isSessionAvailable()).isFalse();
    }

}

File: src/test/kotlin/name/gyger/jmoney/session/SessionServiceTests.kt
package name.gyger.jmoney.session

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner
import javax.transaction.Transactional

@RunWith(SpringRunner::class)
@SpringBootTest
@Transactional
open class SessionServiceTests {

    @Autowired
    lateinit var sessionService: SessionService

    @Autowired
    lateinit var sessionRepository: SessionRepository

    @Test
    fun testBasics() {
        sessionService.initSession()
        assertThat(sessionRepository.isSessionAvailable()).isTrue()

        sessionService.handleContextRefresh(null)
        assertThat(sessionRepository.isSessionAvailable()).isTrue()

        val s = sessionRepository.getSession()
        assertThat(s).isNotNull()
        assertThat(s.rootCategory).isNotNull()
        assertThat(s.splitCategory).isNotNull()
        assertThat(s.transferCategory).isNotNull()
        assertThat(s.id).isNotNull()

        sessionService.removeOldSession()
        assertThat(sessionRepository.isSessionAvailable()).isFalse()
    }

}
